Investigate random fail in `GET /cluster/configuration/validation` endpoint

| Wazuh version | Component |
|-|-|
| 4.3.0 | AIT |

One of the API integration tests failed during a PR check. It only failed once, but we need to investigate why.

## Artifacts

[Build artifacts](https://github.com/wazuh/wazuh/files/7195413/archive.zip)

## Test report

```
_ /tmp/Test_integration_endpoints_B1309_test_files/api/test/integration/test_cluster_endpoints.tavern.yaml::GET /cluster/configuration/validation _
Format variables:
  protocol:s = 'https'
  host:s = 'localhost'
  port:d = '55000'
  test_login_token = 'eyJhbGciOiJFUzUxMiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ3YXp1aCIsImF1ZCI6IldhenVoIEFQSSBSRVNUIiwibmJmIjoxNjMxNjMxMTE4LCJleHAiOjE2MzE2MzIwMTgsInN1YiI6InRlc3RpbmciLCJydW5fYXMiOmZhbHNlLCJyYmFjX3JvbGVzIjpbMV0sInJiYWNfbW9kZSI6IndoaXRlIn0.AJ0hyY1obdmhSppuUXNhIiJQpvtkvgaZkP2cdBekQqWWIq8qIudBRp3-kYv_JDPm9nfz0eRPdLAXinRWqSPorcXqAd0LkcHJrL46Anpy9Krng9yCvMOEEkiI4BUSL6DtNOx51axOhFJwkBbz3OtEy1LGNwJ--l-qq-g3U0euS1kibazX'

Source test stage (line 783):
  - name: Request cluster validation
    request:
      verify: False
      url: "{protocol:s}://{host:s}:{port:d}/cluster/configuration/validation"
      method: GET
      headers:
        Authorization: "Bearer {test_login_token}"
      params:
        nodes_list: 'worker1,worker2'
    response:
      status_code: 200
      json:
        error: !anyint
        data:
          affected_items:
            - name: worker1
              status: 'OK'
            - name: worker2
              status: 'OK'
          failed_items: []
          total_affected_items: 2
          total_failed_items: 0


Formatted stage:
  name: Request cluster validation
  request:
    headers:
      Authorization: Bearer eyJhbGciOiJFUzUxMiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ3YXp1aCIsImF1ZCI6IldhenVoIEFQSSBSRVNUIiwibmJmIjoxNjMxNjMxMTE4LCJleHAiOjE2MzE2MzIwMTgsInN1YiI6InRlc3RpbmciLCJydW5fYXMiOmZhbHNlLCJyYmFjX3JvbGVzIjpbMV0sInJiYWNfbW9kZSI6IndoaXRlIn0.AJ0hyY1obdmhSppuUXNhIiJQpvtkvgaZkP2cdBekQqWWIq8qIudBRp3-kYv_JDPm9nfz0eRPdLAXinRWqSPorcXqAd0LkcHJrL46Anpy9Krng9yCvMOEEkiI4BUSL6DtNOx51axOhFJwkBbz3OtEy1LGNwJ--l-qq-g3U0euS1kibazX
    method: GET
    params:
      nodes_list: worker1,worker2
    url: 'https://localhost:55000/cluster/configuration/validation'
    verify: false
  response:
    json:
      data:
        affected_items:
        - name: worker1
          status: OK
        - name: worker2
          status: OK
        failed_items: []
        total_affected_items: 2
        total_failed_items: 0
      error: !anyint ''
    status_code: 200

Errors:
E   tavern.util.exceptions.TestFailError: Test 'Request cluster validation' failed:
    - List item(s) not present in response: [{'name': 'worker1', 'status': 'OK'}, {'name': 'worker2', 'status': 'OK'}]
------------------------------ Captured log call -------------------------------
WARNING  tavern._plugins.rest.response:response.py:29 Unexpected status code '200'
WARNING  tavern._plugins.rest.response:response.py:29 Unexpected status code '200'
WARNING  tavern._plugins.rest.response:response.py:29 Unexpected status code '200'
WARNING  tavern._plugins.rest.response:response.py:29 Unexpected status code '200'
WARNING  tavern._plugins.rest.response:response.py:29 Unexpected status code '200'
ERROR    tavern.response.base:base.py:41 List item(s) not present in response: [{'name': 'worker1', 'status': 'OK'}, {'name': 'worker2', 'status': 'OK'}]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/tavern/util/dict_util.py", line 406, in check_keys_match_recursive
    assert actual_val == expected_val
AssertionError: assert {'data': {'af...n some nodes'} == {'data': {'af...7f13eb6a59a0>}
  Differing items:
  {'data': {'affected_items': [{'name': 'worker2', 'status': 'OK'}], 'failed_items': [{'error': {'code': 1908, 'message'...diation': 'Please, fix the corrupted files'}, 'id': ['worker1']}], 'total_affected_items': 1, 'total_failed_items': 1}} != {'data': {'affected_items': [{'name': 'worker1', 'status': 'OK'}, {'name': 'worker2', 'status': 'OK'}], 'failed_items': [], 'total_affected_items': 2, 'total_failed_items': 0}}
  {'error': 2} != {'error': <tavern.util.loader.IntSentinel object at 0x7f13eb6a59a0>}
  Left contains 1 more item:
  {'message': 'Could not check v...
  
  ...Full output truncated (33 lines hidden), use '-vv' to show

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/tavern/util/dict_util.py", line 406, in check_keys_match_recursive
    assert actual_val == expected_val
AssertionError: assert {'affected_it...led_items': 1} == {'affected_it...led_items': 0}
  Differing items:
  {'affected_items': [{'name': 'worker2', 'status': 'OK'}]} != {'affected_items': [{'name': 'worker1', 'status': 'OK'}, {'name': 'worker2', 'status': 'OK'}]}
  {'total_affected_items': 1} != {'total_affected_items': 2}
  {'total_failed_items': 1} != {'total_failed_items': 0}
  {'failed_items': [{'error': {'code': 1908, 'message': "Error validating configuration: No rule id or level specified f...e rules: 'etc/rules/new-rules_corrupted.xml'.", 'remediation': 'Please, fix the corrupted files'}, 'id': ['worker1']}]} != {'failed_items': []}
  Full diff:
    {...
  
  ...Full output truncated (26 lines hidden), use '-vv' to show

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/tavern/util/dict_util.py", line 406, in check_keys_match_recursive
    assert actual_val == expected_val
AssertionError: assert [{'name': 'wo...tatus': 'OK'}] == [{'name': 'wo...tatus': 'OK'}]
  At index 0 diff: {'name': 'worker2', 'status': 'OK'} != {'name': 'worker1', 'status': 'OK'}
  Right contains one more item: {'name': 'worker2', 'status': 'OK'}
  Full diff:
  - [{'name': 'worker1', 'status': 'OK'}, {'name': 'worker2', 'status': 'OK'}]
  + [{'name': 'worker2', 'status': 'OK'}]
```

--- a/api/test/integration/common.yaml
+++ b/api/test/integration/common.yaml
@@ -22,6 +22,7 @@ variables:
   restart_delay_cluster: 150
   upgrade_delay: 60
   global_db_delay: 30
+  cluster_sync_delay: 20
 
   custom_wpk_path: "/var/ossec/test_custom_upgrade_3.12.2.wpk"
 

--- a/api/test/integration/test_cluster_endpoints.tavern.yaml
+++ b/api/test/integration/test_cluster_endpoints.tavern.yaml
@@ -16,7 +16,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - name: !anystr
@@ -49,7 +49,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - info:
@@ -80,7 +80,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - info:
@@ -122,7 +122,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - cluster: wazuh
@@ -149,7 +149,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items: &cluster_nodes_response
             - name: !anystr
@@ -169,7 +169,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_nodes_response
@@ -188,7 +188,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_nodes_response
@@ -207,7 +207,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items: []
           failed_items: []
@@ -263,7 +263,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_nodes_response
@@ -282,7 +282,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_nodes_response
@@ -300,7 +300,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_nodes_response
@@ -319,7 +319,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_nodes_response
@@ -351,7 +351,7 @@ stages:
           extra_kwargs:
             select_key: "name,version"
       json:
-        error: !anyint
+        error: 0
         data:
           failed_items: []
           total_affected_items: 3
@@ -370,7 +370,7 @@ stages:
           extra_kwargs:
             select_key: "type"
       json:
-        error: !anyint
+        error: 0
         data:
           failed_items: []
           total_affected_items: 3
@@ -410,7 +410,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - ip: !anystr
@@ -468,7 +468,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           enabled: "yes"
           running: "yes"
@@ -488,7 +488,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - alerts: !anything
@@ -520,7 +520,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - alerts: !anything
@@ -554,7 +554,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - alerts:
@@ -577,7 +577,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - syscheck:
@@ -606,7 +606,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - syscollector:
@@ -635,7 +635,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - alerts:
@@ -657,7 +657,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - syscheck:
@@ -686,7 +686,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - syscollector:
@@ -718,7 +718,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - name: master-node
@@ -743,6 +743,7 @@ stages:
         content-type: application/octet-stream
     response:
       status_code: 200
+    delay_after: !float "{cluster_sync_delay}"
 
   - name: Request validation
     request:
@@ -754,7 +755,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 1
         data:
           failed_items:
             - error:
@@ -765,7 +766,6 @@ stages:
                 - 'worker2'
           total_affected_items: 0
           total_failed_items: 3
-    delay_before: 12
 
     #### Delete corrupted rules file
   - name: Delete rules file
@@ -777,6 +777,7 @@ stages:
         Authorization: "Bearer {test_login_token}"
     response:
       status_code: 200
+    delay_after: !float "{cluster_sync_delay}"
 
   - name: Request cluster validation
     request:
@@ -790,7 +791,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - name: worker1
@@ -800,7 +801,6 @@ stages:
           failed_items: []
           total_affected_items: 2
           total_failed_items: 0
-    delay_before: 12
 
   - name: Request cluster validation
     request:
@@ -814,7 +814,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - name: master-node
@@ -839,7 +839,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - node_name: "master-node"
@@ -896,7 +896,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - node_name: 'worker1'
@@ -924,7 +924,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - global:
@@ -957,7 +957,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - active-response: !anything
@@ -975,7 +975,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - alerts: !anything
@@ -994,7 +994,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - command: !anything
@@ -1012,7 +1012,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - auth:
@@ -1041,7 +1041,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - internal:
@@ -1062,7 +1062,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - internal:
@@ -1094,7 +1094,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - localfile: !anything
@@ -1112,7 +1112,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items: []
           failed_items: []
@@ -1129,7 +1129,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - internal:
@@ -1163,7 +1163,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - monitord:
@@ -1190,7 +1190,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - remote:
@@ -1213,7 +1213,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - internal:
@@ -1254,7 +1254,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - syscheck:
@@ -1273,7 +1273,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - rootcheck:
@@ -1306,7 +1306,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - internal:
@@ -1333,7 +1333,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - wmodules: !anything
@@ -1364,7 +1364,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - path: !anystr
@@ -1402,7 +1402,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items: !anything
           failed_items: []
@@ -1418,7 +1418,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - &cluster_log
@@ -1441,7 +1441,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_log
@@ -1491,7 +1491,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_log
@@ -1511,7 +1511,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_log
@@ -1530,7 +1530,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_log
@@ -1548,7 +1548,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_log
@@ -1569,7 +1569,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - <<: *cluster_log
@@ -1618,7 +1618,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items: !anything
           failed_items: []
@@ -1660,7 +1660,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data: !anything
 
   - name: Cluster stats {node_id} day without stats
@@ -1722,7 +1722,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - total_events_decoded: !anyfloat
@@ -1798,7 +1798,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - averages: !anything
@@ -1830,7 +1830,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - queue_size: !anyfloat
@@ -1868,7 +1868,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - Sun:
@@ -1919,7 +1919,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items: !anything
           failed_items: []
@@ -1943,7 +1943,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - 'worker1'
@@ -1963,7 +1963,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - 'master-node'
@@ -2014,7 +2014,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - alerts:
@@ -2037,7 +2037,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 1
         data:
           affected_items: []
           failed_items:
@@ -2061,7 +2061,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 1
         data:
           affected_items: []
           failed_items:
@@ -2086,7 +2086,7 @@ stages:
     response:
       status_code: 200
       json:
-        error: !anyint
+        error: 0
         data:
           affected_items:
             - alerts:

--- a/api/test/integration/test_cluster_endpoints.tavern.yaml
+++ b/api/test/integration/test_cluster_endpoints.tavern.yaml
@@ -743,7 +743,6 @@ stages:
         content-type: application/octet-stream
     response:
       status_code: 200
-    delay_after: 10
 
   - name: Request validation
     request:
@@ -766,6 +765,7 @@ stages:
                 - 'worker2'
           total_affected_items: 0
           total_failed_items: 3
+    delay_before: 12
 
     #### Delete corrupted rules file
   - name: Delete rules file
@@ -777,7 +777,6 @@ stages:
         Authorization: "Bearer {test_login_token}"
     response:
       status_code: 200
-    delay_after: 10
 
   - name: Request cluster validation
     request:
@@ -801,6 +800,7 @@ stages:
           failed_items: []
           total_affected_items: 2
           total_failed_items: 0
+    delay_before: 12
 
   - name: Request cluster validation
     request:

--- a/api/test/integration/test_cluster_endpoints.tavern.yaml
+++ b/api/test/integration/test_cluster_endpoints.tavern.yaml
@@ -743,6 +743,7 @@ stages:
         content-type: application/octet-stream
     response:
       status_code: 200
+    delay_after: 10
 
   - name: Request validation
     request:
@@ -756,18 +757,15 @@ stages:
       json:
         error: !anyint
         data:
-          affected_items:
-            - name: worker1
-              status: 'OK'
-            - name: worker2
-              status: 'OK'
           failed_items:
             - error:
                 code: 1908
               id:
                 - 'master-node'
-          total_affected_items: 2
-          total_failed_items: 1
+                - 'worker1'
+                - 'worker2'
+          total_affected_items: 0
+          total_failed_items: 3
 
     #### Delete corrupted rules file
   - name: Delete rules file
@@ -779,7 +777,7 @@ stages:
         Authorization: "Bearer {test_login_token}"
     response:
       status_code: 200
-    delay_after: 5
+    delay_after: 10
 
   - name: Request cluster validation
     request:

